Source/WebKit/Platform/IPC/ArgumentCoders.h

```diff
@@ -510,6 +510,67 @@ template<typename T, size_t inlineCapacity, typename OverflowHandler, size_t min
 
 template<typename T, size_t inlineCapacity, typename OverflowHandler, size_t minCapacity> struct ArgumentCoder<Vector<T, inlineCapacity, OverflowHandler, minCapacity>> : VectorArgumentCoder<std::is_arithmetic<T>::value, T, inlineCapacity, OverflowHandler, minCapacity> { };
 
+
+template<bool fixedSizeElements, typename T> struct FixedVectorArgumentCoder;
+
+template<typename T> struct FixedVectorArgumentCoder<false, T> {
+    template<typename Encoder, typename U>
+    static void encode(Encoder& encoder, U&& vector)
+    {
+        static_assert(std::is_same_v<std::remove_cvref_t<U>, FixedVector<T>>);
+
+        encoder << static_cast<size_t>(vector.size());
+        for (auto&& item : vector)
+            encoder << WTF::forward_like<U>(item);
+    }
+
+    template<typename Decoder>
+    static std::optional<FixedVector<T>> decode(Decoder& decoder)
+    {
+        auto size = decoder.template decode<size_t>();
+        if (!size)
+            return std::nullopt;
+
+        // Limit direct allocations from untrusted sources to 1MB.
```

Comment:
It's conservative and safe to match `Vector` in this respect but I suspect not helpful. Is there a chance the `Vector` issue has to do with a specific problem that doesn't exist here?

Reply:
I don't think there is any difference. reserveInitialCapacity vs. FixedVector should be the same from a risk factor.